src/deepsparse/v2/pipeline.py

```diff
@@ -141,23 +144,87 @@ def _run_sub_graphs(
                             inp=operator_output,
                             inference_state=sub_graph.inf,
                             next_step=next_step,
+                            loop=loop,
                         )
                     break
 
             # keep running until all sub graphs have completed.
-            if not any(isinstance(x.output, Future) for x in sub_graphs):
+            if not any(
+                isinstance(x.output, (asyncio.Future, Future)) for x in sub_graphs
+            ):
                 break
 
         return [x.output for x in sub_graphs]
 
-    def _apply_split(self, inp: Any, inference_state: InferenceState):
+    async def run_async(self, *args, inference_state: InferenceState, **kwargs):
         """
-        Split inputs using the pipeline's expand_inputs function. Inputs are split
-        into a batch size of one when a SPLIT_ROUTE node is found in a given pipeline's
-        provided router. The split batches are run asynchronously and then joined when
-        a JOIN_ROUTE node is found, using the pipeline's condense_inputs function.
+        Run through the operators using the provided router and scheduler.
+        The input to a given operator is the output of the previous operator.
+
+        :param inference_state: inference_state for the pipeline.
+        :param pipeline_state: pipeline_state for the pipeline. The values in the state
+            are created during pipeline creation and are read-only during inference.
         """
+        loop = asyncio.get_running_loop()
+
+        next_step = self.router.START_ROUTE
+        operator_output = None
```

Comment:
I'm not sure if there's any reason for an operator to have an output of `None`, but if so, you might consider using another sentinel value here and in the check below 🤷🏻

Reply:
I guess you could also consider adding another local variable to avoid overloading the use of `operator_output`
something like
```suggestion
        operator_output = None
        processed_start_route = False
```